pyccel/codegen/printing/ccode.py

```diff
@@ -306,24 +381,20 @@ def copy_NumpyArray_Data(self, expr):
         lhs = expr.lhs
         if rhs.rank == 0:
             raise NotImplementedError(str(expr))
-        dummy_array_name = self.scope.get_new_name('array_dummy')
-        declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
-        dtype = self.find_in_ndarray_type_registry(self._print(rhs.dtype), rhs.precision)
         arg = rhs.arg if isinstance(rhs, NumpyArray) else rhs
-        if rhs.rank > 1:
-            # flattening the args to use them in C initialization.
-            arg = self._flatten_list(arg)
-
-        self.add_import(c_imports['string'])
         if isinstance(arg, Variable):
-            arg = self._print(arg)
-            cpy_data = "memcpy({0}.{2}, {1}.{2}, {0}.buffer_size);\n".format(lhs, arg, dtype)
-            return '%s' % (cpy_data)
-        else :
-            arg = ', '.join(self._print(i) for i in arg)
-            dummy_array = "%s %s[] = {%s};\n" % (declare_dtype, dummy_array_name, arg)
-            cpy_data = "memcpy({0}.{2}, {1}, {0}.buffer_size);\n".format(self._print(lhs), dummy_array_name, dtype)
-            return  '%s%s' % (dummy_array, cpy_data)
+            return f"array_copy_data({self._print(lhs)}, {self._print(arg)}, 0);\n"
+        order = lhs.order
+        declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
+        dtype = self.find_in_ndarray_type_registry(self._print(rhs.dtype), rhs.precision)
+        array_creations = []
+        dvar = {"dtype": dtype,
+                "declare_dtype": declare_dtype,
+                "order": ('order_c' if order == 'C' else 'order_f'),
```

Comment:
Are there any other possible values for `order`?

Reply:
I am not sure we support the other orders, `K` and `A`, they have different copy rules that do not look very efficient.
@bauom 
[numpy docs](https://numpy.org/doc/stable/reference/generated/numpy.array.html#numpy-array)
[stack answer about order](https://stackoverflow.com/a/51957584)